Added parameters for surface diffusion dependence

angela-moser opened the pull request:
I have added the parameters to allow surface diffusion dependence in the General Rate Model in the cadetAdapter and unitOperation files, which was discussed in [this forum post](https://forum.cadet-web.de/t/adding-surface-diffusion-dependence-for-grm-in-cadet-process/998/2). I copied the format of the original surface diffusivity parameter with the property and setter for all the new parameters, but am not sure if that was the correct way to do this as I am not familiar with these functions in python.

## To do
- [x] Add surface dependence
- [ ] Compare simulation results with "pure" CADET config files (manual test)
- [ ] Add tests (wait until `createLWE` methods are merged into `dev`, i.e. wait until #95 is merged).

schmoelder committed:
Add option to specify (rounding) precision for transforming variables

schmoelder committed:
Expose option to set precision when adding variables

schmoelder committed:
Use variable precision for transformation

schmoelder commented on CADETProcess/processModel/unitOperation.py:
```diff
@@ -896,6 +896,22 @@ class GeneralRateModel(TubularReactorBase):
     surface_diffusion : List of unsigned floats. Length depends on n_comp.
         Diffusion rate for components in adsrobed state.
+    surface_diffusion_dependence : String.
+        Parameter dependence of surface diffusion.    
```
Update docstring (must be one of "LIQUID_...")

schmoelder commented on CADETProcess/processModel/unitOperation.py:
```diff
@@ -1024,7 +1058,119 @@ def surface_diffusion(self, surface_diffusion):
+        return self._surface_diffusion_logkeq_exponent
+    
+    @surface_diffusion_logkeq_exponent.setter
+    def surface_diffusion_logkeq_exponent(self, surface_diffusion_logkeq_exponent):
```
If this one only works with the `Colloidal`, check that type.

schmoelder commented on CADETProcess/processModel/unitOperation.py:
```diff
@@ -1024,7 +1058,119 @@ def surface_diffusion(self, surface_diffusion):
+            
+    @property
+    def surface_diffusion_logkeq_exponent(self):
+        return self._surface_diffusion_logkeq_exponent
```
Generally, consider checking the value of the ` _surface_diffusion_dependence`

angelamoser1 commented:
Soo, I did not mean to add the changes where I added the colloidal binding model to this branch, but git push did not work how I thought it was going to and that's what happened. Since some of the surface diffusivity parameters require having the colloidal binding model anyway and I don't know how to undo this without making an entirely new branch, I am just gonna go with it.

schmoelder committed:
Add support for surface diffusion dependence

angela-moser committed:
Add par_surfdiffusion_dep in adapter and fix colloidal ln(ke) conditional

schmoelder committed:
Fix storage of surface_diffusion in parameters dict

daklauss committed:
Change unittest to pytest

schmoelder committed:
Normalize flow_rate profiles before fitting

daklauss committed:
Remove constructors of testclasses for pytest

daklauss committed:
Add coverage pipeline

ronald-jaepel committed:
Extend MPMLangmuir gamma range

schmoelder committed:
Make run methods private

schmoelder committed:
Call-super().__init__()-for-Simulator-classes

ronald-jaepel committed:
Avoid double transform in _transform

ronald-jaepel committed:
Add functionality to round to significant digits.

schmoelder committed:
Rename TransformBase to TransformerBase

ronald-jaepel committed:
Add warning to simulator when setting n_cycles_max below n_cycles_min

schmoelder committed:
Remove duplicate code

schmoelder committed:
Fix setter for SizedAggregator

schmoelder committed:
Formatting

schmoelder committed:
Return Reaction object when adding to ReactionSystem

schmoelder committed:
Ignore checking size when instance is None

schmoelder committed:
Improve casting of floats

schmoelder committed:
Add tests for Aggregator

schmoelder committed:
Add ProxyList/Array to allow indexed modification of aggregated properties

schmoelder committed:
Improve interpolation of concentration profiles

schmoelder committed:
Formatting, docstrings, and type annotations

schmoelder committed:
Remove Height from Shape metric

schmoelder committed:
Call super().__init__() in FlowSheet

schmoelder committed:
Remove deprecated function

daklauss committed:
Improve fractionation performance

ronald-jaepel committed:
Change warning to Error

ronald-jaepel committed:
Add n_cycles_batch to simulator for batching of cycle calls

AntoniaBerger committed:
Add tool to rename arguments in function signature

Test cover
- basic funcionality
- argument conflict
- rename kwargs function
- with positional arguments
- no arguments

AntoniaBerger committed:
Change indices to components in Reaction.__init__

schmoelder committed:
Add option to update_dict_recursively only for existing keys

schmoelder committed:
Improve docstrings / type annotations

schmoelder committed:
Add tests for recursive_dict module

schmoelder committed:
Add `ChromatographicColumnBase` class

schmoelder committed:
Add some dunder methods

schmoelder committed:
Remove `spatial_method` from discretization parameters

schmoelder committed:
fixup! Add ProxyList/Array to allow indexed modification of aggregated properties

schmoelder committed:
Add method to calculate Bodenstein number

Commit:
Add method to load OptimizationResults from file

schmoelder committed:
Store value of `_q` in parameters dict

schmoelder committed:
Pin libsqlite version to avoid crash

schmoelder committed:
Add dependabot

Fixes #230

ronald-jaepel committed:
Handle nan's in the round_to_significant_digits function

schmoelder committed:
Add concentration to parameters

schmoelder committed:
fixup! Remove `spatial_method` from discretization parameters

schmoelder committed:
Use log transform for optimization variables if ub/lb >= 100

schmoelder committed:
fixup! Add ProxyList/Array to allow indexed modification of aggregated properties

hannahlanzrath committed:
Add use_dll option to LWE tests in test_cadet_adapter

schmoelder committed:
Fix issue when calculating definite integral

schmoelder committed:
Make axial dispersion component dependent

schmoelder committed:
fixup! Remove `spatial_method` from discretization parameters

schmoelder committed:
Improve returning default values for sized parameters

schmoelder committed:
Return n_bounds = 0 if binding_model is NoBinding

schmoelder committed:
Update UnitOperation.q setter

schmoelder committed:
Improve testing of unit operations

schmoelder commented:
Hi @angela-moser it's been some time since worked on this PR. Is this still something you're working on? Should we try to rebase and merge? I'm planning on making a new release in the next week so maybe we can finalize this before?

angela-moser commented:
> Hi @angela-moser it's been some time since worked on this PR. Is this still something you're working on? Should we try to rebase and merge? I'm planning on making a new release in the next week so maybe we can finalize this before?

I think it can be merged. Is there something you need me to do?

schmoelder commented:
> I think it can be merged. Is there something you need me to do?

Have you verified that this in fact produces the same results as manually configured hdf5 files?
Moreover, ideally we would still add unit tests...

angela-moser commented:
> > I think it can be merged. Is there something you need me to do?
> 
> Have you verified that this in fact produces the same results as manually configured hdf5 files? Moreover, ideally we would still add unit tests...

I do not know how to do these things. I would be happy to learn from you if you have the time, but I would not be able to work on this until April.

schmoelder commented:
ok, then let's wait for the next release.
